schema/pgsql/schema.sql

```diff
@@ -2171,6 +2171,27 @@ COMMENT ON COLUMN sla_history_downtime.downtime_id IS 'downtime.id (may referenc
 COMMENT ON COLUMN sla_history_downtime.downtime_start IS 'start time of the downtime';
 COMMENT ON COLUMN sla_history_downtime.downtime_end IS 'end time of the downtime';
 
+CREATE TABLE sla_lifecycle (
+  id bytea20 NOT NULL,
+  environment_id bytea20 NOT NULL,
+  host_id bytea20 NOT NULL,
+  service_id bytea20 DEFAULT NULL,
+
+  -- These columns are nullable, but as we're using the delete_time to build the composed primary key, we have to set
+  -- this to `0` instead, since it's not allowed to use a nullable column as part of the primary key.
+  create_time biguint NOT NULL DEFAULT 0,
+  delete_time biguint NOT NULL DEFAULT 0,
+
+  CONSTRAINT pk_sla_lifecycle PRIMARY KEY (id, delete_time)
+);
```

Comment:
I was thinking about creating some indices for the queries within the `SyncCheckablesSlaLifecycle` function.

In a huge environment, those queries may take some time. However, I cannot say how huge they must become to make those queries slow or if Icinga 2 will be the performance bottleneck before.

I have not tested this or anything, just wanted to put the idea out there.

Reply:
> or if Icinga 2 will be the performance bottleneck before

If there's an issue, you could probably trigger it without Icinga 2 becoming an issue. You can create and delete many hosts and services over time, keeping the active config in Icinga 2 at a constant size but growing this table.